### PR TITLE
nixos/cassandra: Add cfg.extraEnvSh

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Allow extending `cassandra-env.sh`, as is required when extending the classpath. This can be used for custom logging.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by roberth** [comment]
@cransom could you have a look please?

File: nixos/modules/services/databases/cassandra.nix [hunk 1]
```diff
@@ -38,13 +38,18 @@ let
       cassandraYaml = builtins.toJSON cassandraConfigWithAddresses;
       cassandraEnvPkg = "${cfg.package}/conf/cassandra-env.sh";
       cassandraLogbackConfig = pkgs.writeText "logback.xml" cfg.logbackConfig;
+      passAsFile = [ "extraEnvSh" ];
+      inherit (cfg) extraEnvSh;
```
Comment (JohnAZoidberg) [inline nixos/modules/services/databases/cassandra.nix:42]: This is necessary for `passAsFile`? Or could you do `passAsFile = [ "cfg.extraEnvSh" ];`?

**Review by JohnAZoidberg — COMMENTED** [review]
Looks good. Given that you provided an example, I assume you have tested it?

File: nixos/modules/services/databases/cassandra.nix (hunk 1 above)
Comment (roberth) [reply nixos/modules/services/databases/cassandra.nix:42]: Yes, those two lines pretty much sum up how it works.

It's [documented](https://nixos.org/manual/nix/stable/#sec-advanced-attributes) in the Nix manual if you'd like to know more.

**Comment by roberth** [comment]
> I assume you have tested it?

Yes I have and it works.

**Review by JohnAZoidberg — APPROVED** [review]

File: nixos/modules/services/databases/cassandra.nix (hunk 1 above)
Comment (JohnAZoidberg) [reply nixos/modules/services/databases/cassandra.nix:42]: Ahh, right! The second line is required to make it an environment variable in the first place.

**Comment by roberth** [comment]
Thanks @JohnAZoidberg !

**Pull request merged**